backend/models/user.js

```diff
@@ -74,12 +91,57 @@ const findOne = async function (query, projection) {
 	return await DB.findOne("admin", COLL_NAME, query, projection);
 };
 
+const handleAuthenticateFail = async function (username) {
+	const currentTime = Date.now();
+
+	try {
+		const user = await User.findByUserName(username);
+		const elapsedTime = user.customData.lastFailedLoginAt ?
+			currentTime - user.customData.lastFailedLoginAt : undefined;
+
+		const failedLoginCount = user.customData.failedLoginCount && elapsedTime && elapsedTime < config.lockoutDuration ?
+			user.customData.failedLoginCount + 1 : 1;
+
+		const loginLocked = failedLoginCount >= config.maxUnsuccessfulLoginAttempts ? true : undefined;
+
+		await User.update(username, {
+			"customData.lastFailedLoginAt": currentTime,
+			"customData.failedLoginCount": failedLoginCount,
+			"customData.loginLocked": loginLocked
```

Comment:
I don't think we should be storing this? we should always deduce it at runtime?

Also doesn't look like we use it

Reply:
We don't use it. It is always deduced at runtime. But this was put in to satisfy: 
> Account locking triggers a flag which can be monitored on either Elastic or Intercom (or perhaps both)

Reply:
let's remove it, considering the states can't be guaranteed it's pointless to store it. It's not like Elastic/Intercom has the ability to probe the database anyway, what we actually want to do is to send a signal to them to notify them this account has been locked out when it happens.

@mosment3DRepo if you want this to be sent to intercom you need to tell us how you want it done, do we update the user data?

@keifgwinn can you advise charence on what he needs to send?

Reply:
@carmenfan can we post to the events service [Intercom API docs here](https://developers.intercom.com/intercom-api-reference/reference#submitting-events)

Endpoint: https://api.intercom.io/events

JSON would be
```
{
  "event_name": "Password Lockout",
  "created_at": 1613668686,
  "user_id": "example_user"
}
```

Reply:
@mosment3DRepo Done! Event name is `password-lockout`.